library/core/src/macros/mod.rs

```diff
@@ -198,14 +198,11 @@ pub macro assert_matches {
 
 /// A macro for defining `#[cfg]` match-like statements.
 ///
-/// It is similar to the `if/elif` C preprocessor macro by allowing definition of a cascade of
-/// `#[cfg]` cases, emitting the implementation which matches first.
+/// The `cfg_select!` macro checks a series of `#[cfg]` conditions, and picks the branch that
+/// belongs to the first condition that evaluates to true.
```

Comment:
```suggestion
/// This macro evaluates, at compile-time, a series of `cfg` predicates, selects
/// the first that is true, and emits the code guarded by that predicate. The
/// code guarded by other predicates is not emitted.
```

We call these "configuration predicates" in the Reference, so I'd suggest keeping with that.  This should also mention what it does after picking a predicate, i.e. that it emits the code.  Probably we should mention that we're doing this at compile-time and that the code behind the other predicates is not emitted.